src/w3iProxy/authProviders/internalAuthProvider.ts

```diff
@@ -1,12 +1,13 @@
 import { getAccount, watchAccount, watchNetwork, getNetwork } from '@wagmi/core'
 import type { JsonRpcRequest } from '@walletconnect/jsonrpc-types'
 import type { EventEmitter } from 'events'
+import { getChainString } from '../../utils/chain'
 
 export default class InternalAuthProvider {
   private readonly methodsListenedTo = ['auth_set_account']
   public providerName = 'InternalAuthProvider'
-  public account?: string
-  public chain?: string
+  public account?: string = getAccount().address
```

Comment:
We can initialize the state with the default values from the Wagmi. This is the main thing that will solve the wallet flashing issue. We could get the auth status in the first render, which is awesome.